src/file_handler.rs

```diff
@@ -123,22 +124,126 @@ fn config_dir_path() -> String {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use file_handler::ConfigManagement;
+    use file_handler::FileManagement;
     use file_handler::FileSystem;
+    use std::fs;
     use std::io;
 
     struct MockFileSystem;
+    struct MockFileHandler;
 
     impl FileSystem for MockFileSystem {
         fn create_dir(&self, _path: &str) -> io::Result<()> {
-            Ok(())
+            fs::create_dir_all(&_path)
+        }
+    }
+
+    impl ConfigManagement for MockFileHandler {
+        fn config_dir_create(&self) -> io::Result<String> {
+            fs::create_dir_all(config_dir_path()).expect("Cannot create directory");
+            Ok(config_dir_path())
+        }
+
+        fn config_dir_exists(&self) -> bool {
+            self.file_exists(&config_dir_path())
+        }
+
+        fn config_read(&self, _file: ConfigFile) -> io::Result<String> {
+            let _config_file_path = config_path(_file);
```

Comment:
The `_variable` style is usually used to 1) indicate private variables in non-typed environments (like python, javascript) or 2) to indicate that the variable is not being used.

Since we're using this variable I think we should just call it `config_file_path`. Same goes for all other uses of `_variable` that are being used.